Sources/AsyncKit/EventLoopFuture/EventLoopFutureQueue.swift

```diff
@@ -0,0 +1,99 @@
+import NIO
+
+/// Allows you to queue closures that produce an `EventLoopFuture`, so each future only gets run if the previous ones complete, succeed, or fail.
+public final class EventLoopFutureQueue {
+
+    /// Under what conditions an appended closure should be run.
+    public enum Continue {
+
+        /// Run closure on the previous future's success.
+        case success
+
+        /// Run closure on the previous future's failure.
+        case failure
+
+        /// Run closure on the previous future's completion.
+        case complete
+    }
+
+    /// Errors that get propogated based on a future's completion status and the next appended closure's continuation condition.
+    public enum ContinueError: Error {
+
+        /// A previous future failed with an error, which we don't desire.
+        case previousError(Error)
+
+        /// A previous future succeeded, which we don't desire.
+        case previousSuccess
```

Comment:
This might cause a bunch of headaches with the generic type system - but what if we also provided the value that succeeded that we didn't expect? It'd make the asynchronous debugging a bit nicer

Reply:
The best way to do this would probably be to pass in the value as `Any`, and then potentially the event loop that the value came from. I can't think of a good way to make that generic.

Reply:
The only way I see is to make the entire queue generic, but don't think that's a great idea.